tests/validation/tests/v3_api/test_configmaps.py

```diff
@@ -0,0 +1,506 @@
+
+def rbac_configmap_delete(p_client, ns):
+
+    # Verify creating, editing and deleting config maps is functional.
+    # The p_client passed as the parameter would be as per the role assigned
+    value = "valueall"
+    keyvaluepair = {"testall": value}
+    configmap = create_configmap(keyvaluepair, p_client, ns)
+
+    # Verify deletion of config map
+    delete_configmap(p_client, configmap, ns, keyvaluepair)
+
+
+@pytest.fixture(scope='module', autouse="True")
+def create_project_client(request):
+    client, cluster = get_user_client_and_cluster()
+    create_kubeconfig(cluster)
+    p, ns = create_project_and_ns(USER_TOKEN, cluster, "testconfigmap")
+    p_client = get_project_client_for_token(p, USER_TOKEN)
+    c_client = get_cluster_client_for_token(cluster, USER_TOKEN)
+
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+    namespace["c_client"] = c_client
+
+    def fin():
+        client = get_user_client()
+        client.delete(namespace["project"])
+    request.addfinalizer(fin)
+
+
+def validate_workload_with_configmap(p_client, workload,
+                                     type, ns_name, keyvaluepair,
+                                     workloadwithconfigmapasvolume=False,
+                                     workloadwitconfigmapasenvvar=False,
+                                     podcount=1):
+
+    validate_workload(p_client, workload, type, ns_name, pod_count=podcount)
+
+    pod_list = p_client.list_pod(workloadId=workload.id).data
+    mountpath = "/test"
+    for i in range(0, len(keyvaluepair)):
+        key = list(keyvaluepair.keys())[i]
+        if workloadwithconfigmapasvolume:
+            key_file_in_pod = mountpath + "/" + key
+            print(key_file_in_pod)
+            command = "cat " + key_file_in_pod + ''
+            print(" Command to display configmap value from container is: ")
+            print(command)
+            result = kubectl_pod_exec(pod_list[0], command)
+            assert result.decode("utf-8") == (list(keyvaluepair.values())[i])
+        elif workloadwitconfigmapasenvvar:
+            command = 'env'
+            result = kubectl_pod_exec(pod_list[0], command)
+            print(list(keyvaluepair.values())[i])
+            if list(keyvaluepair.values())[i] in result.decode("utf-8"):
+                assert True
+
+
+def delete_configmap(client, configmap, ns, keyvaluepair):
+
+    key = list(keyvaluepair.keys())[0]
+    print("Delete Configmap")
+    client.delete(configmap)
+
+    # Sleep to allow for the configmap to be deleted
+    time.sleep(5)
+    print("Config Map list after deleting config map")
+    configmapdict = client.list_configMap()
+    print(configmapdict)
+    print(configmapdict.get('data'))
+    if len(configmapdict.get('data')) > 0:
+        testdata = configmapdict.get('data')
+        print("TESTDATA")
+        print(testdata[0]['data'])
+        if key in testdata[0]['data']:
```

Comment:
Why do we only check the data part of the first item (which index is 0) in `testdata`? is the position fixed?